### PR TITLE
Ability to propagate custom options to kubelet

### PR DESCRIPTION
**Pull request by ytsarev**
Main motivation for us is to utilize
https://kubernetes.io/docs/concepts/cluster-administration/kubelet-garbage-collection/

Anything from https://kubernetes.io/docs/admin/kubelet/
can be potentially used as a value

### CODE REVIEW

**Comment by mumoshu** [comment]
@ytsarev Thank you very much for your contribution 👍 
Not saying this shouldn't be merged but just curious - what would be pros of exposing this via `cluster.yaml`?
Like DOCKER_OPTS passed to dockerd, we've been recommending users to edit and version-control files generated by kube-aws so that they can make all the kinds of changes possible via not only cluster.yaml but cloud-configs and stack templates.
Do you find exposing it via cluster.yaml better for some reasons?

**Comment by ytsarev** [comment]
@mumoshu actually we intentionally never modify cloud formation templates. We version control them , but modification go strictly through `kube-aws render stack`. This way we avoid any ad-hoc modifcations and possible incompatibility with a tool. I believe such approach brings reliability I would like to stick with it.

**Comment by mumoshu** [comment]
@ytsarev Thanks, I understand the motivation 👍

**Comment by mumoshu** [comment]
Hey @redbaron, WDYT on this?

I believe we can't prioritize these kinds of improvements over other scalability, availability, reliability things we're focusing on - creating a cloudformation.yaml isn't our focus :wink:
However, I also believe that we can accept contributions of these kinds as long as it doesn't badly affect our main focus. I understand @ytsarev's motivation on this and it doesn't seem to affect our focus badly. It is useful and orthogonal to our focus.

So, I'm wanting this to be merged. WDYT?

File: core/controlplane/config/templates/cluster.yaml [hunk 1]
```diff
@@ -468,6 +468,12 @@ worker:
 #      kube2IamSupport:
 #        enabled: true
 #
+#      # Propagate custom CLI options to kubelet
+#      # Provided example enables docker image garbage collection as in https://kubernetes.io/docs/concepts/cluster-administration/kubelet-garbage-collection/
```
Comment (redbaron) [inline core/controlplane/config/templates/cluster.yaml]: is it correct? Isn't GC enabled by default, just with different thresholds?

**Comment by redbaron** [comment]
@ytsarev , there is an issue about git workflow where it is possible to keep your customizations and continue receiving updates from kube-aws: https://github.com/kubernetes-incubator/kube-aws/issues/462

@mumoshu , PR looks allright, cluster.yaml already grew beyond point of no return :)

File: core/controlplane/config/templates/cluster.yaml (hunk 1 above)
Comment (ytsarev) [inline core/controlplane/config/templates/cluster.yaml]: @redbaron definitely not enabled by default, at least not explicitly

Comment (redbaron) [inline core/controlplane/config/templates/cluster.yaml]: From your link:

```
User Configuration
Users can adjust the following thresholds to tune image garbage collection with the following kubelet flags :

image-gc-high-threshold, the percent of disk usage which triggers image garbage collection. Default is 90%.

image-gc-low-threshold, the percent of disk usage to which image garbage collection attempts to free. Default is 80%.
```

Comment (ytsarev) [inline core/controlplane/config/templates/cluster.yaml]: ok, then they are too high by default ,right. Want me to change formulation in the comment ?

Comment (redbaron) [inline core/controlplane/config/templates/cluster.yaml]: yes please. we have many people starting playing with kube using kube-aws, providing accurate information is important. Original formulation gives impression that GC is switched off, which might cause panic :)

File: core/controlplane/config/templates/cloud-config-worker [hunk 2]
```diff
@@ -622,7 +622,7 @@ write_files:
     permissions: 0755
     owner: root:root
     content: |
-      KUBELET_OPTS=""
+      KUBELET_OPTS="{{.Experimental.KubeletOpts}}"
```
Comment (redbaron) [inline core/controlplane/config/templates/cloud-config-worker]: I think both controller and worker kubelets should be using KubeletOpts

Comment (ytsarev) [inline core/controlplane/config/templates/cloud-config-worker]: Indeed, i will amend it

**Comment by mumoshu** [comment]
@redbaron 

> cluster.yaml already grew beyond point of no return :)

True :rofl:

File: core/controlplane/config/templates/cloud-config-worker (hunk 2 above)
Comment (ytsarev) [inline core/controlplane/config/templates/cloud-config-worker]: @redbaron PR amended, please review

**Comment by redbaron** [comment]
Looks good now, thanks for your contribution!

**Pull request merged**